decidim-accountability/spec/forms/admin/result_import_projects_form_spec.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim::Accountability
+  describe Admin::ResultImportProjectsForm do
+    subject(:form) { described_class.from_params(attributes).with_context(context) }
+
+    let(:organization) { create(:organization) }
+    let(:participatory_process) { create :participatory_process, organization: organization }
+    let(:current_component) { create :accountability_component, participatory_space: participatory_process }
+    let(:budget_component) { create(:component, manifest_name: "budgets", participatory_space: participatory_process) }
+    let(:import_all_selected) { false }
+
+    let(:context) do
+      {
+        current_organization: organization,
+        current_component: current_component,
+        current_participatory_space: participatory_process
+      }
+    end
+
+    let(:attributes) do
+      {
+        origin_component_id: budget_component.id,
+        import_all_selected_projects: import_all_selected
+      }
+    end
+
+    describe "when origin component presents" do
+      let(:import_all_selected) { true }
+
+      it { is_expected.to be_valid }
+    end
+
+    describe "when import all is not selexted" do
+      let(:import_all_selected) { false }
+
+      it { is_expected.not_to be_valid }
+    end
+
+    describe "when origin component does not present" do
+      let(:origin_component_id) { nil }
+
+      it { is_expected.not_to be_valid }
+    end
+
+    describe "#origin_components" do
+      subject { described_class.from_model(current_component).with_context(context) }
+      before do
+        create_list(:component, 4, manifest_name: "budgets", participatory_space: participatory_process)
+      end
+
+      it "returns all of the components" do
+        expect(subject.origin_components&.ids).to match_array(Decidim::Component.where(manifest_name: "budgets")&.ids)
```

Comment:
Don't use the `&` method calls in tests to hide potential errors in the functionality.

E.g. if you are expecting the `origin_component` to exist in the test, remove the `&.` at the end of it. If you are not expecting it to exist, write a test which checks that.